### PR TITLE
Include ROM API symbols

### PR DESCRIPTION
**Pull request by bjoernQ**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

#### Description

Since users seem to struggle with including the `rom_functions.x` from esp-wifi we now include everything needed in esp-hal which shouldn't hurt. Hopefully also the process of updating is a bit easier now.

Once this PR landed there will be a follow up PR to remove the linker scripts from esp-wifi-sys.

`skip-changelog` because of changes in esp-hal

#### Testing
Examples still work

### CODE REVIEW

**Review by jessebraham — APPROVED** [review]
Changes LGTM, thanks for taking care of this!

**Comment by bugadani** [comment]
> skip-changelog because of changes in esp-hal

To be fair I'm pretty sure that "No need to include `rom_functions.x` manually" belongs to the changelog :D

**Comment by bjoernQ** [comment]
> > skip-changelog because of changes in esp-hal
> 
> To be fair I'm pretty sure that "No need to include `rom_functions.x` manually" belongs to the changelog :D That, or the migration guide is in the wrong crate :)

mhhhh - but that applies to esp-wifi not esp-hal (and it's mentioned in esp-wifi's changelog)

**Comment by bugadani** [comment]
But the migration guide you edited applies to esp-hal 🙃

**Comment by bjoernQ** [comment]
> But the migration guide you edited applies to esp-hal 🙃

oops - yeah that's the wrong crate

**Comment by bjoernQ** [comment]
`skip-changelog` again - see above

File: examples/build.rs [hunk 1]
```diff
@@ -1,14 +1,4 @@
 fn main() {
-    if cfg!(feature = "esp32c6") || cfg!(feature = "esp32h2") {
-        println!("cargo::rustc-link-arg=-Trom_coexist.x");
-        println!("cargo::rustc-link-arg=-Trom_functions.x");
-        println!("cargo::rustc-link-arg=-Trom_phy.x");
-    }
-
-    if cfg!(feature = "esp-wifi") {
```
Comment (bugadani) [inline examples/build.rs:8]: Would you remove this from hil-tests, too, please?

Comment (bjoernQ) [reply examples/build.rs:8]: 👍 forgot we had it there, too

**Review by MabezDev — APPROVED** [review]
Thanks!

**Pull request merged**